src/components/ha-expansion-panel.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import { fireEvent } from "../common/dom/fire_event";
+import "./ha-svg-icon";
+import { mdiChevronDown } from "@mdi/js";
+import { classMap } from "lit-html/directives/class-map";
+
+@customElement("ha-expansion-panel")
+class HaExpansionPanel extends LitElement {
+  @property({ type: Boolean, reflect: true }) expanded = false;
+
+  @property({ type: Boolean, reflect: true }) outlined = false;
+
+  protected render(): TemplateResult {
+    return html`
+      <div class="summary" @click=${this._toggleContainer}>
+        <slot name="title"></slot>
+        <ha-svg-icon
+          .path=${mdiChevronDown}
+          class="summary-icon ${classMap({ expanded: this.expanded })}"
+        ></ha-svg-icon>
+      </div>
+      <div
+        class="container ${classMap({ expanded: this.expanded })}"
+        @transitionend=${this._handleTransitionEnd}
+      >
+        <slot></slot>
+      </div>
+    `;
+  }
+
+  private _handleTransitionEnd() {
+    const container = this.shadowRoot!.querySelector(
+      ".container"
+    )! as HTMLDivElement;
+
+    if (this.expanded) {
+      container.style.height = "auto";
+    }
+  }
+
+  private _toggleContainer(): void {
+    const container = this.shadowRoot!.querySelector(
+      ".container"
+    )! as HTMLDivElement;
+
+    const scrollHeight = container.scrollHeight;
+    container.style.height = `${scrollHeight}px`;
```

Comment:
Is this also needed for the closing transition?

Reply:
yes, because when expanded I have `height: auto`. I need to set explicit height before I start the transition.